### PR TITLE
Final step of adding create_mount_from_config

### PR DESCRIPTION
**Pull request by ImmortalRabbit**
_create_mount is deleted, a new arg Mount is added, not used unit tests were fixed.

## Related Issue

#873 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: pocs/observatory.py [hunk 1]
```diff
@@ -53,12 +50,8 @@ def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
 
         # TODO(jamessynge): Discuss with Wilfred the serial port validation behavior
         # here compared to that for the mount.
-        self.logger.info('\tSetting up dome.')
```
Comment (wtgee) [inline pocs/observatory.py]: Let's keep these log statements but change to "Addng dome to Observatory" or something similar.

File: pocs/observatory.py [hunk 2]
```diff
@@ -53,12 +50,8 @@ def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
...
         self.dome = dome
-
-        self.logger.info('\tSetting up scheduler')
```
Comment (wtgee) [inline pocs/observatory.py]: Keep this as well, as per above.

File: pocs/observatory.py [hunk 3]
```diff
@@ -39,10 +39,7 @@ def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
         self.observer = None
         self._setup_location()
 
-        self.logger.info('\tSetting up mount')
```
Comment (wtgee) [inline pocs/observatory.py]: Keep statement but change to adding, as mentioned below.

File: pocs/observatory.py [hunk 4]
```diff
@@ -23,7 +23,7 @@
 
 class Observatory(PanBase):
 
-    def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
+    def __init__(self, cameras=None, scheduler=None, dome=None, mount=None, *args, **kwargs):
```
Comment (wtgee) [inline pocs/observatory.py]: Updates to the docstring are needed. Can either be here or separate PR to get all of them.

File: pocs/tests/test_observatory.py [hunk 5]
```diff
@@ -65,23 +67,6 @@ def test_bad_site(simulator, config):
...
-@pytest.mark.without_mount
-def test_bad_mount_driver(config):
-    conf = config.copy()
-    simulator = hardware.get_all_names(without=['mount'])
-    conf['mount']['driver'] = 'foobar'
-    with pytest.raises(error.MountNotFound):
-        Observatory(simulator=simulator, config=conf, ignore_local_config=True)
-
-
 def test_can_observe(config, caplog):
```
Comment (wtgee) [inline pocs/tests/test_observatory.py]: Should this test be named `test_cannot_observe`? And if so we should say the reason why (e.g. `test_no_mount_cannot_observe`)

File: pocs/tests/test_observatory.py [hunk 6]
```diff
@@ -34,9 +34,11 @@ def observatory(config, simulator, images_dir):
     site_details = create_location_from_config(config)
     scheduler = create_scheduler_from_config(config, observer=site_details['observer'])
     dome = create_dome_from_config(config)
+    mount = create_mount_from_config(config)
```
Comment (wtgee) [inline pocs/tests/test_observatory.py]: You probably need `config_with_simulated_mount` instead of just `config`.

File: pocs/tests/test_observatory.py [hunk 7]
```diff
@@ -34,9 +34,11 @@ def observatory(config, simulator, images_dir):
...
     obs = Observatory(config=config,
                       scheduler=scheduler,
                       dome=dome,
+                      mount=mount,
                       simulator=simulator,
```
Comment (wtgee) [inline pocs/tests/test_observatory.py]: This `simulator` is no longer used and should be removed.

File: pocs/tests/test_observatory.py (hunk 6 above)
Comment (wtgee) [reply pocs/tests/test_observatory.py]: Doing this makes it work for me:

```
@pytest.fixture
def observatory(config_with_simulated_mount, images_dir):
    """Return a valid Observatory instance with a specific config."""
    config = config_with_simulated_mount
```

File: conftest.py [hunk 8]
```diff
@@ -12,12 +12,14 @@
 import subprocess
 import time
 
+
```
Comment (wtgee) [inline conftest.py]: Extra 
```suggestion
```
Extra blank line.

File: conftest.py [hunk 9]
```diff
@@ -12,12 +12,14 @@
...
 from pocs import hardware
 from pocs.utils.database import PanDB
 from pocs.utils.logger import get_root_logger
 from pocs.utils.messaging import PanMessaging
 
 # Global variable set to a bool by can_connect_to_mongo().
+
```
Comment (wtgee) [inline conftest.py]:
```suggestion
```
Extra blank line.

File: pocs/observatory.py [hunk 10]
```diff
@@ -39,10 +37,8 @@ def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
         self.observer = None
         self._setup_location()
 
-        self.logger.info('\tSetting up mount')
-        self.mount = None
-        self._create_mount()
-
+        self.logger.info('\tAdding mount to Observatory.')
```
Comment (wtgee) [inline pocs/observatory.py]: Not needed and inaccurate. If no mount is passed this message would still be generated.
```suggestion
```

File: pocs/observatory.py [hunk 11]
```diff
@@ -39,10 +37,8 @@ def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
...
+        self.mount = mount
```
Comment (wtgee) [inline pocs/observatory.py]: Use the method that you created, which will also generate the appropriate log message.

```suggestion
        self.set_mount(mount)
```

File: pocs/observatory.py [hunk 12]
```diff
@@ -53,12 +49,10 @@ def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
 
         # TODO(jamessynge): Discuss with Wilfred the serial port validation behavior
         # here compared to that for the mount.
-        self.logger.info('\tSetting up dome.')
+        self.logger.info('\tAdding dome to Observatory.')
```
Comment (wtgee) [inline pocs/observatory.py]: Same as mount comment. 
```suggestion
```

File: pocs/observatory.py [hunk 13]
```diff
@@ -53,12 +49,10 @@ def __init__(self, cameras=None, scheduler=None, dome=None, *args, **kwargs):
...
         self.dome = dome
-
-        self.logger.info('\tSetting up scheduler')
+        self.logger.info('\tAdding scheduler to Observatory.')
```
Comment (wtgee) [inline pocs/observatory.py]: Same as mount comment.
```suggestion
```

File: pocs/tests/test_pocs.py [hunk 14]
```diff
@@ -51,18 +52,23 @@ def scheduler(config):
...
-def observatory(config, db_type, cameras, scheduler, dome):
+def mount(config_with_simulated_mount):
+    return create_mount_from_config(config_with_simulated_mount)
+
+
+@pytest.fixture(scope='function')
+def observatory(config_with_simulated_mount, db_type, cameras, scheduler, dome, mount):
     observatory = Observatory(
-        config=config,
+        config=config_with_simulated_mount,
```
Comment (wtgee) [inline pocs/tests/test_pocs.py]: Since you are passing in the created mount simulator do you still need this config?

File: pocs/tests/test_pocs.py [hunk 15]
```diff
@@ -265,57 +292,55 @@ def wait_for_message(sub, type=None, attr=None, value=None):
...
-        # Insert a dummy weather record to break wait
-        observatory.db.insert_current('weather', {'safe': True})
-
-        assert wait_for_state(msg_subscriber, 'scheduling')
-    finally:
-        cmd_publisher.send_message('POCS-CMD', 'shutdown')
-        pocs_thread.join(timeout=30)
-
-    assert pocs_thread.is_alive() is False
+#
```
Comment (wtgee) [inline pocs/tests/test_pocs.py]: This test cannot be commented out.

File: pocs/tests/test_pocs.py [hunk 16]
```diff
@@ -104,6 +110,25 @@ def pocs_with_dome(config_with_simulated_dome, db_type, dome):
     pocs.power_down()
 
 
+@pytest.fixture(scope='function')
+def pocs_with_mount(config_with_simulated_mount, db_type, mount):
```
Comment (wtgee) [inline pocs/tests/test_pocs.py]: This should just be `pocs` rather than `pocs_with_mount` as having a mount would be the default. Need to change all instances below as well.

Comment (ImmortalRabbit) [reply pocs/tests/test_pocs.py]: > This should just be `pocs` rather than `pocs_with_mount` as having a mount would be the default. Need to change all instances below as well.

I have added pocs_with_mount because `test_power_down_while_running` requires to have `mount` but not `dome`. So, `pocs_with_mount` is used to have a `mount` only without the `dome`.

Comment (ImmortalRabbit) [reply pocs/tests/test_pocs.py]: > This should just be `pocs` rather than `pocs_with_mount` as having a mount would be the default. Need to change all instances below as well.

I have added pocs_with_mount because `test_power_down_while_running` requires to have `mount` but not `dome`. So, `pocs_with_mount` is used to have a `mount` only without the `dome`.

File: pocs/tests/test_pocs.py [hunk 17]
```diff
@@ -385,8 +390,7 @@ def test_power_down_while_running(pocs):
     assert pocs.connected is False
 
 
-def test_power_down_dome_while_running(pocs_with_dome):
```
Comment (wtgee) [inline pocs/tests/test_pocs.py]: You need to add these two lines back in for the three failing tests and then they should pass.

File: pocs/tests/test_pocs.py [hunk 18]
```diff
@@ -75,6 +80,7 @@ def pocs(config, observatory):
 
     pocs = POCS(observatory,
                 run_once=True,
+                mount=mount,
```
Comment (wtgee) [inline pocs/tests/test_pocs.py]: `POCS` class doesn't need a mount, it's passed via the `observatory` instance.

```suggestion
```

File: pocs/tests/test_pocs.py [hunk 19]
```diff
@@ -84,18 +90,19 @@ def pocs(config, observatory):
...
-                              simulator=simulator,
                               dome=dome,
+                              mount=mount,
                               ignore_local_config=True,
                               db_type=db_type
                               )
 
     pocs = POCS(observatory,
                 run_once=True,
+                dome=dome,
```
Comment (wtgee) [inline pocs/tests/test_pocs.py]: Not needed.
```suggestion
```

File: pocs/tests/test_pocs.py [hunk 20]
```diff
@@ -84,18 +90,19 @@ def pocs(config, observatory):
...
+                mount=mount,
```
Comment (wtgee) [inline pocs/tests/test_pocs.py]:
```suggestion
```
